### DISCUSSION TITLE
How do I query a foreign table if the foreign keys are in an array?

### DISCUSSION THREAD

**Discussion by 0xDebabrata** [supabase/supabase#2430] [Questions] [answered]
I have two tables 'cards' and 'tags'. 'cards' has a column which contains an array of tag ids that each card has. Is there a way I could fetch all the tags a card has in a single query?

For example the following query fetches all cards but for the tags it returns the array of tag ids that card has. So I would need to make separate queries that fetch the required tags.

```javascript
        const { data:cards, error } = await supabase
            .from("cards")
            .select(`
                excerpt,
                note,
                created_at,
                tags,
                collections (
                    id,
                    name,
                    author
                )
            `)

```
It should be like accessing a foreign table as done with the 'collections' table in the above query. But the problem is that since tags column of 'cards' contains an array, I cannot do that.

Do I need to change my table structure or is there any better option?

**Comment by steve-chavez** [comment, marked as answer]
> Do I need to change my table structure or is there any better option?

Yes, changing the table structure would be the best option. To query foreign tables you need a foreign key defined, and adding a foreign key on arrays is not supported in PostgreSQL as previously discussed in https://github.com/supabase/supabase/discussions/535

As mentioned on the previous discussion, you'll need an additional junction table(say `card_tag`) and then you'll be able to join cards with tags as per your snippet above.

**Reply by 0xDebabrata** [reply]
Thanks for the suggestion. I will set this up then.
